File: exercises/pi/pi.py
import math
from numba import njit, prange

steps = 100000

@njit(parallel=True)
def pi(steps):
    pi = math.acos(-1.0)
    dx = 1./steps

    lsum = 0.0
    for i in prange(0, steps):
        x = (i-0.5)*dx
        lsum = lsum + 4./(1.0 + x*x)

    lsum = lsum * dx
    return lsum

lsum = pi(steps)
print(lsum)

File: exercises/pi/pi.f90
program pi_serial

  implicit none


  integer, parameter::steps=1000000000
  integer i
  double precision dx,x,sum,pi
  
  pi=acos(-1.d0)
  dx=1.d0/dble(steps)
  sum=0.d0

  !$omp parallel private (i, x) shared(dx)
  !$omp do reduction(+: sum)
  do i=1,steps
     x=(dble(i)-.5d0)*dx
     sum=sum+4.d0/(1.d0+x*x)
  enddo
  !$omp end do
  !$omp end parallel

  sum=sum*dx

  print *,sum,abs(pi-sum)/pi

end program pi_serial
